### ISSUE TITLE
Fluctuating accuracy in down stream tasks

### ISSUE THREAD

**Issue by Harika-BV** [KeremTurgutlu/self_supervised#65]
Hey @KeremTurgutlu , while performing down stream tasks with SWAV according to your tutorial (https://keremturgutlu.github.io/self_supervised/04%20-%20training_swav_iwang.html#Downstream-Task ), we found fluctuations in the accuracy (ups and downs) and its not getting plateaued
Any reason for this?

 
![image](https://user-images.githubusercontent.com/34688053/126765004-8fb8acc3-a4c7-42b1-aa0f-f91a215fb047.png)

Thanks

**Comment by KeremTurgutlu** [comment]
Swav is used for pretraining your encoder, so you can think it as a very powerful weight initialization :) Once you have that encoder rest is classical transfer learning. Usually the recommended way of finetuning is to freeze the earlier layers and only train remaining, then unfreeze every layer and train everything with lower learning rate so that you have a better convergence. 

If I was getting such results I would ask the following questions for debugging:

- First thing would be to figure out whether pretraining went well, e.g. is the model able to learn good representations? You can check this in multiple ways; if you have _some_ labelled data you can try clustering or knn based retrieval to see if your representation/embeddings from the same classes are close to each other. I assume you have some labelled data since you mention a downstream task. Besides retrieval based approaches you can make sure that pretraining training loss and validation loss follow a similar downtrend during training, e.g. you don't want to overfit or let your representations collapse (issues like all embeddings being same).

- If you are sure that pretraining was successful, then debugging downstream is as same as debugging any other transfer learning approach. Lowest hanging fruit is to set a good learning rate, maybe you picked too high? What happens when you don't use pretrained weights and start from random weights? If same issue happens then you know it's not related to pretraining.

There are bunch of reasons for training to fail. Here is a great article from Andrej Karpathy: http://karpathy.github.io/2019/04/25/recipe/
